src/schema/event.ts

```diff
@@ -1,5 +1,41 @@
+import { EventStatus } from "@prisma/client";
 import { z } from "zod";
 
+import prisma from "@/lib/prisma";
+
+const stringOrArray = z.union([z.string(), z.array(z.string())]);
+
+const isValidEventStatus = (value: string[]): value is EventStatus[] => {
+  return value.every((status) => status in EventStatus);
+};
+
+/**
+ * Splits all strings by their comma delimitter into one large array
+ */
+const transformStringOrArray = (value: string | string[]) =>
+  (typeof value === "string"
+    ? value.length == 0
+      ? []
+      : value.split(",").map((item) => item.trim())
+    : value.flatMap((item) => item.split(",").map((item) => item.trim()))) ||
+  [];
+
```

Comment:
Next.js request should allow multiple values for a given query parameter without having to use commas to seperate the values. For example, `/api/event?eventType="123"&eventType="abc"` should return an array of strings for eventType in `req.query`.

Additionally, you can use 
```javascript
eventType: z.preprocess((input) => {
		return typeof input === 'string' || input instanceof String ? [input] : input;
	}, z.string().array())
	.optional(),
``` 
to transform single-value query parameters into an array. For example, `/api/event?eventType="123"` will return eventType as a string, but using z.preprocess, you can convert that to an array of strings for eventType with a single value inside that array.

Reply:
Yep, the code I have written is mostly for taking into account how we are passing params with the URLSearchParams object. It doesn't directly accept an array, so you can make it pass multiple params where next js will read it as an array like:
```ts
// this can be simplified into a foreach loop
let params = new URLSearchParams();
params.append('arrayparams', 'val1');
params.append('arrayparams', 'val2');
params.append('arrayparams', 'val3');
```
I decided it's easier if we just convert the array of strings to a string by joining the array with `.join(",")`

Although thinking about it, if an Event Type does end up having a comma, that will break it. I'll redo it to take multiple params and allow commas 👍

Reply:
Went with something slightly different since if the API is called with a param that is empty (`param=`) then it will be treated as an empty array, but is read as an empty string which would get converted to [""] causing some issues.

```ts
const toArray = <T>(value: T | T[]) => {
  if (typeof value === "string" && value.trim() === "") {
    return [] as T[]; // Return an empty array for an empty string
  }

  return Array.isArray(value) ? value : [value];
};
```